### DISCUSSION TITLE
Ensemble of different NN models?

### DISCUSSION THREAD

**Discussion by cosmicnet** [TorchEnsemble-Community/Ensemble-Pytorch#104] [General]
Hello,

I'm looking to build an ensemble of different PyTorch models. I can across your module, and from what I can tell (without running any code) the ensembles are based on the same NN model design? The ensembles are also tied to the training process (in that, to make predictions you start with untrained models that must go through the training process).

I was initially planning on building an ensemble from some pre-trained models I have which are based on different NN designs. This was to see what results I would get from different combinations of the models.

I'm assuming that if I look through the code I could probably find a way of using Ensemble-Pytorch to achieve what I want, rather than rolling my own code.

Maybe someone could offer some advice?

**Comment by ibromodzi** [comment]
Did you manage to find any way around this?  I'm looking into implementing something similar to what you described.

**Reply by cosmicnet** [reply]
I ended up working on some separate things so never go around to implementing. However I may be circling back to it this coming year. Let me know how you get on.

**Comment by ibromodzi** [comment]
Oh, sure. I'll look into it one of these days.
